rsa.go

```diff
@@ -120,10 +120,29 @@ func VerifyRSASignature(publicKey, signature, data []byte, digestType string, pk
 	}
 
 	if pkeyopt != nil && len(pkeyopt) > 0 {
-		for k, v := range pkeyopt {
-			if C.X_EVP_PKEY_CTX_ctrl_str(ctx, C.CString(k), C.CString(v)) <= 0 {
-				return false, fmt.Errorf("failed to set %s", k)
+		// This is a convenience function for calling X_EVP_PKEY_CTX_ctrl_str. The _Ctype_struct_evp_pkey_ctx_st type is not
+		// exposed, but ctx can be captured in a local function like this.
+		setKeyOpt := func(pkeyopt map[string]string, k string) error {
+			v, ok := pkeyopt[k]
+			if !ok {
+				return nil
+			}
+			ck := C.CString(k)
+			defer C.free(unsafe.Pointer(ck))
+			cv := C.CString(v)
+			defer C.free(unsafe.Pointer(cv))
+			if C.X_EVP_PKEY_CTX_ctrl_str(ctx, ck, cv) <= 0 {
+				return fmt.Errorf("failed to set %s", k)
 			}
+			return nil
+		}
+
+		// Set RSA padding mode if it exists. Order matters; mode must be set before salt length.
+		if err := setKeyOpt(pkeyopt, "rsa_padding_mode"); err != nil {
+			return false, err
+		}
```

Comment:
If `rsa_padding_mode` does not exist and `rsa_pss_saltlen` does exist, it will return the `failed to set rsa_pss_saltlen` error.  That's expected behavior, but it might be valuable to return an error explaining that?

Reply:
Okay, I changed the logic so that it will only set `rsa_pss_saltlen` if `rsa_padding_mode` is already set to pss.